include/micm/util/vector_matrix.hpp

```diff
@@ -55,7 +100,7 @@ namespace micm
       {
         if (other.size() < y_dim_)
         {
-          throw std::runtime_error("Matrix row size mismatch in assignment from vector.");
+          throw std::system_error(make_error_code(MicmVectorMatrixErrc::RowSizeMismatch), "");
```

Comment:
Might be worth having a message why it's mismatched for easier debugging?

Reply:
done!